### DISCUSSION TITLE
Modify action in listener middleware

### DISCUSSION THREAD

**Discussion by mauriceoc** [reduxjs/redux-toolkit#3629] [General]
Just wondering if there's a way to safely modify an action in `redux-toolkit` listeners?

It seems from the doc like they only respond to actions. 

```
middleware.startListening({
     matcher: myMatcher,
     effect: (action, api) => {
           // modify the action somehow here
     }
});
```

Is custom middleware required in order to modify actions?

**Comment by EskiMojo14** [comment]
Listeners are called after the action already reached reducers, so it's not possible to intercept and replace the action.

You would need to write a custom middleware for that :)
